autoscale/src/main/java/com/sequenceiq/periscope/controller/DistroXAutoScaleClusterV1Controller.java

```diff
@@ -119,6 +119,9 @@ private DistroXAutoscaleClusterResponse createClusterJsonResponse(Cluster cluste
     }
 
     private DistroXAutoscaleClusterResponse updateClusterAutoScaleState(Cluster cluster, AutoscaleClusterState autoscaleState) {
+        if (autoscaleState.getUseStopStartMechanism() == null) {
+            autoscaleState.setUseStopStartMechanism(asClusterCommonService.shouldEnableStopStartBasedOnEntitlement(cluster));
+        }
```

Comment:
There are two entitlement checks happening in this section of the code. One here and another a few lines below - `alertValidator.validateStopStartEntitlementAndDisableIfNotEntitled(cluster);`. I think both these entitlement checks can be integrated instead of making separate UMS API calls.

Reply:
Than, here we can directly set it to true if it is null and not change it if not null.
Something like this right:-
if (autoscaleState.getUseStopStartMechanism() == null) {
            autoscaleState.setUseStopStartMechanism(true);
        }